streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -305,6 +297,12 @@ public void resume() {
             default:
                 throw new IllegalStateException("Illegal state " + state() + " while resuming active task " + id);
         }
+        if (requiresUpdate) {
+            partitionGroup.updatePartitions(inputPartitions(), recordQueueCreator::createQueue);
+            if (state() != State.RESTORING) { // if task is RESTORING then topology will be initialized in completeRestoration
```

Comment:
IIUC we are trying to avoid calling the initializeTopology twice, I'm thinking whether this optimization is necessary, as the guarantee to call  `initializeTopology` in RESTORING state is weak, which might be implicitly changed in the future, cc @guozhangwang

Reply:
when work on this task it fails some test that ensure that initializeTopology called once here. I thought that it might be important and decided to support this.